### PR TITLE
subscriber: "implementing FormatEvent" docs 

### PR DESCRIPTION
**Pull request by hawkw**
This branch adds some documentation to the `FormatEvent` trait in
`tracing_subscriber::fmt` on how to write user-provided `FormatEvent`
implementations. There's probably room for additional improvement here,
but I just wanted to get something written down for now.

I also fixed a broken link I noticed while I was here.

### CODE REVIEW

File: tracing-subscriber/src/fmt/format/mod.rs [hunk 1]
```diff
@@ -35,15 +35,55 @@ use fmt::{Debug, Display};
...
+///
+///   The [`Writer`] type also provides additional methods that provide
+///   information about how the event should be formatted. The
+///   [`Writer::has_ansi_escapes`] method indicates whether [ANSI terminal
+///   escape codes] are supported by the underlying I/O writer that the event
+///   will be written to. If this returns `true`, the formatter is permitted to
+///   use ANSI escape codes to add colors and other text formatting to its
+///   output. If it returns `false`, the event will be written to an output that
+///   does not support ANSI escape codes (such as a log file), and they should
+///   not be emitted.
```
Comment (davidbarsky) [inline tracing-subscriber/src/fmt/format/mod.rs:75]:
```suggestion
///   The [`Writer`] type also implements additional methods that provide
///   information about how the event should be formatted. The
///   [`Writer::has_ansi_escapes`] method indicates whether [ANSI terminal
///   escape codes] are supported by the underlying I/O writer that the event
///   will be written to. If this returns `true`, the formatter is permitted to
///   use ANSI escape codes to add colors and other text formatting to its
///   output. If it returns `false`, the event will be written to an output that
///   does not support ANSI escape codes (such as a log file), and they should
///   not be emitted.
```

File: tracing-subscriber/src/fmt/format/mod.rs [hunk 2]
```diff
@@ -35,15 +35,55 @@ use fmt::{Debug, Display};
...
+///
+///   Crates like [`ansi_term`] and [`owo-colors`] can be used to add ANSI
```
Comment (davidbarsky) [inline tracing-subscriber/src/fmt/format/mod.rs:77]: lol, that's a good name

File: tracing-subscriber/src/fmt/mod.rs [hunk 3]
```diff
@@ -223,7 +223,44 @@
...
 //!
-//! ### Filters
+//! ### Customizing Formatters
+//!
+//! The formatting of log lines for spans and events is controlled by two
+//! traits, [`FormatEvent`] and [`FormatFields`]. The [`FormatEvent`] trait
+//! determines the overall formatting of the log line, such as what information
+//! from the event's metadata and span context is included and in what order.
+//! The [`FormatFields`] trait determines how fields &mdash; both the event's
+//! fields and fields on spans &mdash; are formatted.
```
Comment (davidbarsky) [inline tracing-subscriber/src/fmt/mod.rs:233]: is the use of `&mdash;` instead of `—` intentional?

**Review by davidbarsky — APPROVED** [review]

File: tracing-subscriber/src/fmt/mod.rs (hunk 3 above)
Comment (hawkw) [reply tracing-subscriber/src/fmt/mod.rs:233]: i can't remember the hotkey for typing a literal em dash on my keyboard...

**Pull request merged**